### ISSUE TITLE
Question about Work Profile

### ISSUE THREAD

**Issue by rahaaatul** [iamr0s/Dhizuku#21]
Can this API be used to create work Profile? Can apps like [Island](https://github.com/oasisfeng/island)/[Insular](https://gitlab.com/secure-system/Insular)/[Shelter](https://gitea.angry.im/PeterCxy/Shelter) utilize this API to create and manage work profiles?

**Comment by iamr0s** [comment]
Maybe not

**Comment by rahaaatul** [comment]
What would be your suggestion if someone wants to have work profile and have device owner permissions? if any.

**Comment by lelemka0** [comment]
Profile can be created when `DISALLOW_ADD_MANAGED_PROFILE` is disabled, which is provided by `Install-Lion` in DPM mode.

> https://github.com/dadaewq/Install-Lion/blob/cc0b8badbb7ee34c670d280c5c05e3d8d82eec4a/app/src/main/res/xml/pref_dpm_settings.xml#L15C36-L15C36
> https://github.com/dadaewq/Install-Lion/blob/cc0b8badbb7ee34c670d280c5c05e3d8d82eec4a/app/src/main/java/com/modosa/apkinstaller/fragment/DpmSettingsFragment.java#L45C24-L45C24

```
devicePolicyManager.addUserRestriction(adminComponentName, UserManager.DISALLOW_ADD_MANAGED_PROFILE);
devicePolicyManager.clearUserRestriction(adminComponentName, UserManager.DISALLOW_ADD_MANAGED_PROFILE);
```
Use the following apk to easily modify it:
  https://github.com/lelemka0/DPM_Operate/actions/runs/6601951179

With this option turned off (May need reboot), `Shelter` can create the workspace correctly. 
Workspace with `Island` can only be created manually because the app checks the device owner and refuses to continue.

This method only applies to API26-29. For Android 11 and above, I didn't find a feasible way because the feature has been removed by aosp.

> This constant was deprecated in API level 30.
> As the ability to have a managed profile on a fully-managed device has been removed from the platform, this restriction will be silently ignored when applied by the device owner.

Reference:
https://developer.android.com/reference/android/os/UserManager#DISALLOW_ADD_MANAGED_PROFILE
https://stackoverflow.com/a/72330509

**Comment by fu8765** [comment]
@lelemka0 your dpm app doesn't work. I install it but cannot find it anywhere somehow.

**Comment by lelemka0** [comment]
> @lelemka0 your dpm app doesn't work. I install it but cannot find it anywhere somehow.

Please confirm whether the installation is successful. Because the ci version is not signed, it is impossible to install successfully without core patch.

**Comment by fu8765** [comment]
> > @lelemka0 your dpm app doesn't work. I install it but cannot find it anywhere somehow.
> 
> Please confirm whether the installation is successful. Because the ci version is not signed, it is impossible to install successfully without core patch.

No, i don't think the install is successful. There is no response after installing the apk, nor a fair message, but i can't locate the package afterwards

**Comment by dxxxxp** [comment]
Could a enterprise API be used in any FOSS fashion? Just a thought. Another thought, As I recall there existed a NFC function for profile ownership.

**Comment by dxxxxp** [comment]
I updated my above comment.

**Comment by WAYYYAW** [comment]
> My solution to this has been to set dhizuku as profile owner instead of device owner 和 at the very least Hail seems to be working, since the hide mode works. I would like to know if dhizuku is impaired in any way through this approach outside of not being able to grant any access to apps in the work profile.


It might be due to my device. If dhizuku is set as the profile owner, it cannot revoke this permission through ADB, and even dhizuku itself cannot be deactivated. Worse still, it will not be possible to set device owner permissions for other apps.

**Comment by s1204IT** [comment]
The basic premise is that if either the device owner or profile owner exists, it is not possible to set both to different apps due to the Android specifications.

This is easier to understand if you imagine a typical usage scenario.
As the name suggests, the device owner owns the device itself, so all profiles on the device are defined as common owned profiles for the organization.
Profile Owner is different and is only used when a user owns a device and only some features are provided to the organization (e.g. BYoD).

And in both situations, they cannot coexist because they are already provisioned.

**Comment by s1204IT** [comment]
The conditions for disabling via ADB are device admin and debug version.

**Comment by jachk96** [comment]
Means this issue still can't be solved up until today (2025)?
